README.md

```diff
@@ -11,3 +11,28 @@ repository is just a convenience for experiments and prototypes.
 See [SECURITY/SBOM Software Bill of Materials](https://cwiki.apache.org/confluence/display/SECURITY/SBOM+Software+Bill+of+Materials)
 for a quick background on what SBOMs are and how they're produced and
 used across the ASF.
+
+## Repository structure
+
+### `sboms`
+
+This folder contains an archive of SBOMs downloaded from their canonical
+locations. The top-level directory is the PMC, under that the directory
+structure follows the
+[Purl](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst)
+identification of the artifacts that are described by the SBOMs.
```

Comment:
given SBOMs don't always come from a type-related repository (Maven Central is an exception, but others come from a website or from download ares), you can't know what will be the type at download time
the fact that most projects publish SBOMs to Maven Central is causing a misconception: look at other ones and you'll see

Reply:
> I agree we should be careful not to be too Maven-focused

and even in "Maven", there is a distinction between the repository (Maven Central only, AFAIK, but we n=may have surprises in the future) and the build tools (Maven, Gradle, sbt, Ant, and whatever will be chosen by ASF projects to build their JVM-oriented code, that is not even only Java but also Kotlin, Scala, Groovy, ...)

`maven` as a simple id carries much lack of clarity: it has so many different meanings in different contexts

Reply:
> TBH, you can't really know the type based on analysis either

what the SBOM says is what the SBOM producer says: if the project generates content that does not represent its reality (even unintentionally), it's a bug, that should be reported and fixed. SBOMs are about that too, as part of benefits from transparency

that's why for the storage structure, it's safer to describe real facts before adding any layer of interpretation: we know for a real fact where it was published as we downloaded it from there. And a project may publish SBOMs in multiple places in parallel (and in the worst case with different data, based on the time of generation...)

anything else is much less guaranteed 100% accurate

Reply:
another thought:
it's clear that Airflow is much about Python, and **uses** pypi for its dependencies: it uses pypi for downloading external dependencies, but does it publish its own internal dependencies to pypi? I don't know.

I don't know details of Airflow project, but I know the equivalent in the JVM ecosystem: not everything is published to Maven Central, only dependencies that need to be consumed

orienting SBOMs only against where dependencies are published and consumed is IMHO a restrictive view that should be avoided

(again, the fact that SBOMs open transparency discussion is really a good thing, whatever our conclusion will be: this discussion is wider that the only technical conclusion that we'll solidify as directory names in Git)

Reply:
> if the project generates content that does not represent its reality (even unintentionally), it's a bug, that should be reported and fixed.

Of course!

> not everything is published to Maven Central, only dependencies that need to be consumed

That is definitely true.

> orienting SBOMs only against where dependencies are published and consumed is IMHO a restrictive view that should be avoided

I agree we should not orient SBOMs against where dependencies are published - or downloaded. I think we should orient SBOMs against the artifacts that they describe. That means we need identifiers for those artifacts - which we need for many other purposes as well. Purl currently seems like the most promising mechanism for identifiers for artifacts (though they are not without their challenges either). While this might get somewhat messy once we'll need Purls with qualifiers and subpaths, I think using Purls still makes most sense.

Reply:
purl is oriented toward where dependencies are published, because this is the way it solves the coordinates question: coordinates are necessary when you publish/consume

> I think we should orient SBOMs against the artifacts that they describe

I'd love too, that would be ideal, but I'm afraid it's not possible

in fact, this was my first learning when I saw SBOMs from Airflow and I discovered how they split 1 release in 21 SBOMs = 5 Python versions * 4 SBOM + 1 `npm-only`

always having coordinates is one bias we have based on Maven Central experience, and I did not get we had that bias before you started this study: thanks for it, I was desperately looking for such diverse consolidation :)